sdk/src/main/scala/com/horizen/account/forger/AccountForgeMessageBuilder.scala

```diff
@@ -121,37 +118,37 @@ class AccountForgeMessageBuilder(
             val (txBaseFeePerGas, txForgerTipPerGas) = GasUtil.getTxFeesPerGas(ethTx, baseFeePerGas)
             cumBaseFee = cumBaseFee.add(txBaseFeePerGas.multiply(txGasUsed))
             cumForgerTips = cumForgerTips.add(txForgerTipPerGas.multiply(txGasUsed))
-
+            priceAndNonceIter.next()
           case Failure(e: GasLimitReached) =>
             // block gas limit reached
             // keep trying to fit transactions into the block: this TX did not fit, but another one might
             log.trace(s"Could not apply tx, reason: ${e.getMessage}")
             // skip all txs from the same account
-            accountsToSkip += tx.getFrom
+            priceAndNonceIter.pop()
```

Comment:
seems a bit tricky that `pop` excludes all the entries for given account.
Maybe we may choose another name

Reply:
OK, I'll change the name of pop method to removeAndSkipAccount.